### PR TITLE
mb-707 add partitions to athena based on year and month

### PR DESCRIPTION
**Pull request by rdhariwal**
## Description

Athena queries have been slowing down due to lack of partitioning.  This PR extends the table creation logic and adds partitioning capability so we can run queries like:

```
SELECT actions_executed,
         elb_status_code,
         *
FROM alb_logs
WHERE elb_status_code = 500
        AND month =12
        AND year=2019 limit 100
```

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make build_tools
# delete table alb_logs
query-lb-logs -v true

```

Then navigate to [athena](https://us-west-2.console.aws.amazon.com/athena/home?region=us-west-2#query) in aws and run 
```
SHOW Partitions `alb_logs`;
```

## Code Review Verification Steps

* [x] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/tree/master/docs/backend.md#logging)
* [x] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [x] Request review from a member of a different team.
* [x] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-707?atlOrigin=eyJpIjoiODVlMGU0OGIyYjYzNGM5YTk4OWNjMDNiZjliNjgxYjciLCJwIjoiaiJ9) for this change
* [additional documentation](https://github.com/transcom/ppp-infra/pull/786)

## Screenshots
image

### CODE REVIEW

**Comment by rdhariwal** [comment]
This tool has been ran for experimental, staging and prod. So all logs are partitioned by month and year in all environments.

**Comment by rdhariwal** [comment]
> Just to clarify, is it implied that if the alb table exists it will be in partitions already? If already exists and not in partitions do is there a manual process?

That is correct, tables in athena are just scaffolding to query the data from the bucket. So my view is they are throwaway. So I am leaning towards drop and re-create them and partitioning made sense as part of setup. Have any concerns Lee?

**Comment by mr337** [comment]
> So I am leaning towards drop and re-create them and partitioning made sense as part of setup. Have any concerns Lee?

Nope just wanted to see if we run this the query will magically take care of partitioning. If not, which I believe I understand that it wont, we may need some docs :)

**Comment by rdhariwal** [comment]
> > So I am leaning towards drop and re-create them and partitioning made sense as part of setup. Have any concerns Lee?
> 
> Nope just wanted to see if we run this the query will magically take care of partitioning. If not, which I believe I understand that it wont, we may need some docs :)

👍 I can also make a query param that will just add partitions. I already have an `if exists` clause in partition creation. Documenting either makes sense. Let me take care of that!

**Comment by rdhariwal** [comment]
> 🚀 - I'm approving but I'd love for you to have more doc updates in this PR.

more docs here: https://github.com/transcom/ppp-infra/pull/786

File: cmd/query-lb-logs/main.go [hunk 1]
```diff
@@ -243,10 +247,22 @@ func CheckEnv(serviceAthena *athena.Athena, logger, infoLogger *log.Logger, v *v
 
 	if !isTableFound {
 		logger.Println("Table not found, creating table....")
-		err := createLogTable(serviceAthena, logger, infoLogger, dbName, logBucket)
+		err = createLogTable(serviceAthena, logger, infoLogger, dbName, logBucket)
```
Comment (chrisgilmerproj) [inline cmd/query-lb-logs/main.go]: This seems to be changing the value of `err` outside the scope. I am not sure you wanted to do that. Maybe a better setup here is to be specific with the error name so you don't trigger the shadow variables linter. 

```suggestion
		errCreateLogTable := createLogTable(serviceAthena, logger, infoLogger, dbName, logBucket)
```

**Comment by rdhariwal** [comment]
> 🚀 - this looks good but I was curious about one bit of code that looked odd to me. Can you double check?

refactored err variable usage. I went back and forth on it and landed on avoiding re-using the var when possible is a good practice.

**Pull request merged**